core/src/test/java/com/novoda/noplayer/mediaplayer/AndroidMediaPlayerFacadeTest.java

```diff
@@ -0,0 +1,482 @@
+        argumentCaptor.getValue().onPrepared(mediaPlayer);
+
+        verify(preparedListener).onPrepared(mediaPlayer);
+    }
+
+    @Ignore("Should we throw when a listener is not attached?")
+    @Test
+    public void givenNoBoundPreparedListener_andMediaPlayerIsPrepared_whenPrepared_thenDoesNotForwardOnPrepared() {
+
+    }
+
+    @Test
+    public void givenBoundVideoSizeChangedListener_andMediaPlayerOnPrepared_whenVideoSizeChanges_thenForwardsSizeChanges() {
+        givenMediaPlayerIsPrepared();
+
+        ArgumentCaptor<MediaPlayer.OnVideoSizeChangedListener> argumentCaptor = ArgumentCaptor.forClass(MediaPlayer.OnVideoSizeChangedListener.class);
+        verify(mediaPlayer).setOnVideoSizeChangedListener(argumentCaptor.capture());
+        argumentCaptor.getValue().onVideoSizeChanged(mediaPlayer, ANY_WIDTH, ANY_HEIGHT);
+
+        verify(videoSizeChangedListener).onVideoSizeChanged(eq(mediaPlayer), eq(ANY_WIDTH), eq(ANY_HEIGHT));
+    }
+
+    @Ignore("Should we throw when a listener is not attached?")
+    @Test
+    public void givenNoBoundVideoSizeChangedListener_andMediaPlayerIsPrepared_whenVideoSizeChanged_thenDoesNotForwardOnSizeChanges() {
+
+    }
+
+    @Test
+    public void givenBoundCompletionListener_andMediaPlayerIsPrepared_whenCompleted_thenForwardsCompleted() {
+        givenMediaPlayerIsPrepared();
+
+        ArgumentCaptor<MediaPlayer.OnCompletionListener> argumentCaptor = ArgumentCaptor.forClass(MediaPlayer.OnCompletionListener.class);
+        verify(mediaPlayer).setOnCompletionListener(argumentCaptor.capture());
+        argumentCaptor.getValue().onCompletion(mediaPlayer);
+
+        verify(completionListener).onCompletion(mediaPlayer);
+    }
+
+    @Ignore("Should we throw when a listener is not attached?")
+    @Test
+    public void givenNoBoundCompletionListener_andMediaPlayerIsPrepared_whenCompleted_thenDoesNotForwardCompleted() {
+
+    }
+
+    @Test
+    public void givenBoundErrorListener_andMediaPlayerIsPrepared_whenErroring_thenForwardsError() {
+        givenMediaPlayerIsPrepared();
+
+        ArgumentCaptor<MediaPlayer.OnErrorListener> argumentCaptor = ArgumentCaptor.forClass(MediaPlayer.OnErrorListener.class);
+        verify(mediaPlayer).setOnErrorListener(argumentCaptor.capture());
+        argumentCaptor.getValue().onError(mediaPlayer, ANY_ERROR_WHAT, ANY_ERROR_EXTRA);
+
+        verify(errorListener).onError(mediaPlayer, ANY_ERROR_WHAT, ANY_ERROR_EXTRA);
+    }
+
+    @Ignore("Should we throw when a listener is not attached?")
+    @Test
+    public void givenNoBoundErrorListener_andMediaPlayerIsPrepared_whenErroring_thenDoesNotForwardError() {
+
+    }
+
+    @Test
+    public void givenBoundBufferListener_andMediaPlayerIsPrepared_whenBuffering_thenBufferPercentageIsUpdated() {
+        givenMediaPlayerIsPrepared();
+
+        ArgumentCaptor<MediaPlayer.OnBufferingUpdateListener> argumentCaptor = ArgumentCaptor.forClass(MediaPlayer.OnBufferingUpdateListener.class);
```

Comment:
What about wrapping this into a method `givenMediaPlayerIsBuffering` (for symmetry with the prepared one) ?